go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -96,100 +93,80 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
+	if err != nil {
+		klog.Errorf("Unable to list egress firewall ACLs, cannot cleanup old stale data, err: %v", err)
+		return
+	}
+
 	if config.Gateway.Mode == config.GatewayModeShared {
-		// Mode is shared gateway mode, make sure to delete all ACLs on the node switches
-		egressFirewallACLIDs, stderr, err := util.RunOVNNbctl(
-			"--data=bare",
-			"--no-heading",
-			"--columns=_uuid",
-			"--format=table",
-			"find",
-			"acl",
-			fmt.Sprintf("priority<=%s", types.EgressFirewallStartPriority),
-			fmt.Sprintf("priority>=%s", types.MinimumReservedEgressFirewallPriority),
-		)
-		if err != nil {
-			klog.Errorf("Unable to list egress firewall logical router policies, cannot cleanup old stale data, stderr: %s, err: %v", stderr, err)
-			return
-		}
-		if egressFirewallACLIDs != "" {
-			nodes, err := oc.watchFactory.GetNodes()
-			if err != nil {
-				klog.Errorf("Unable to cleanup egress firewall ACLs remaining from local gateway mode, cannot list nodes, err: %v", err)
-				return
-			}
-			logicalSwitches := []string{}
-			for _, node := range nodes {
-				logicalSwitches = append(logicalSwitches, node.Name)
-			}
-			for _, logicalSwitch := range logicalSwitches {
-				switchACLs, stderr, err := util.RunOVNNbctl(
-					"--data=bare",
-					"--no-heading",
-					"--columns=acls",
-					"--format=table",
-					"list",
-					"logical_switch",
-					logicalSwitch,
-				)
-				if err != nil {
-					klog.Errorf("Unable to remove egress firewall acl, cannot list ACLs on switch: %s, stderr: %s, err: %v", logicalSwitch, stderr, err)
-				}
-				for _, egressFirewallACLID := range strings.Fields(egressFirewallACLIDs) {
-					if strings.Contains(switchACLs, egressFirewallACLID) {
-						_, stderr, err := util.RunOVNNbctl(
-							"remove",
-							"logical_switch",
-							logicalSwitch,
-							"acls",
-							egressFirewallACLID,
-						)
-						if err != nil {
-							klog.Errorf("Unable to remove egress firewall acl: %s on %s, cannot cleanup old stale data, stderr: %s, err: %v", egressFirewallACLID, logicalSwitch, stderr, err)
+
+		// Mode is shared gateway mode, make sure to delete all egfw ACLs on the node switches
+		if len(egressFirewallACLs) != 0 {
+			for _, egressFirewallACL := range egressFirewallACLs {
+
+				nodeLogicalSwitchLookup := func(item *nbdb.LogicalSwitch) bool {
+					// Ignore external and Join switches
+					if strings.Contains(item.Name, "join") || strings.Contains(item.Name, "ext") {
+						return false
+					}
+
+					for _, acl := range item.ACLs {
+						if acl == egressFirewallACL.UUID {
+							return true
 						}
 					}
+					return false
 				}
+
+				nodeSwitches, err := libovsdbops.FindSwitch(oc.modelClient.GetClient(), nodeLogicalSwitchLookup)
+				if err != nil {
+					klog.Errorf("Error while querying ACLs uuid:%s with reject action: %s", egressFirewallACL.UUID, err)
+					return
+				}
+
+				err = libovsdbops.RemoveACLFromSwitches(oc.modelClient, nodeSwitches, egressFirewallACL.UUID)
```

Comment:
I would probably have all this within a `libovsdbops.RemoveACLsFromNodeSwitches`. You can probably not worry about finding which ACL is on which specific switch, but if you do, you can also do a single iteration through the cache having a map of switch to ACLs that you can use to delete after.